ApplicationLibCode/FileInterface/RifFaultReactivationModelExporter.cpp

```diff
@@ -240,30 +266,53 @@ std::pair<bool, std::string>
 //--------------------------------------------------------------------------------------------------
 ///
 //--------------------------------------------------------------------------------------------------
-std::pair<bool, std::string> RifFaultReactivationModelExporter::printMaterials( std::ostream& stream )
+std::pair<bool, std::string>
+    RifFaultReactivationModelExporter::printMaterials( std::ostream&                                                   stream,
+                                                       const RimFaultReactivationModel&                                rimModel,
+                                                       const std::map<RimFaultReactivation::ElementSets, std::string>& materialNames )
 {
     // MATERIALS PART
     struct Material
     {
         std::string name;
         double      density;
-        double      elastic1;
-        double      elastic2;
+        double      youngsModulus;
+        double      poissonNumber;
         double      permeability1;
         double      permeability2;
     };
 
     RifInpExportTools::printSectionComment( stream, "MATERIALS" );
+    std::vector<Material> materials;
+
+    for ( auto [element, materialName] : materialNames )
+    {
+        std::array<double, 3> parameters = rimModel.materialParameters( element );
+
+        // Incoming unit for Young's Modulus is GPa: convert to Pa.
+        double youngsModulus = RiaEclipseUnitTools::gigaPascalToPascal( parameters[0] );
+
+        // Poisson Number has no unit.
+        double poissonNumber = parameters[1];
+
+        // Unit is already kg/m^3
+        double density = parameters[2];
+
+        materials.push_back( Material{ .name          = materialName,
```

Comment:
Here is a good example on how readable the code is when using designated initializer. Nice!